viewing_party/party.py

```diff
@@ -0,0 +1,156 @@
+    user_data["watchlist"].append(movie)
+    return user_data
+
+#Function to move movie from user's watchlist to watched list
+def watch_movie(user_data, movie_title):
+    for movie in user_data["watchlist"]:
+        for key, movie_tit in movie.items():
+            if movie_tit == movie_title:
+                user_data["watched"].append(movie)
+                user_data["watchlist"].remove(movie)       
+    return user_data
+
+# Test wave 02 functions
+
+#Function returns avg rating from user's watched list
+def get_watched_avg_rating(user_data):
+    
+    avg = 0
+    if user_data["watched"]:
+        ratings = [movie["rating"] for movie in user_data["watched"]]
+        avg = sum(ratings) / len(ratings)
+    return avg
+
+
+
+#Function returns most frequent genre from watched list
+
+def get_most_watched_genre(user_data):
+    
+    if user_data["watched"]:
+
+        #Create a list of all watched genres
+        watched_genres = [movie["genre"] for movie in user_data["watched"]]
+
+        #Create a frequency dictionary for all watched genres
+        freq = {}
+        for genre in watched_genres:
+            freq[genre] = watched_genres.count(genre)
+
+        #finding the most watched genre and accounting for the edge case of multiple genres tying
+        max_genre_list = []
+        max_genre = 0
+        for genre in set(watched_genres):
+            if freq[genre] > max_genre:
+                max_genre_list = [genre]
+                max_genre = freq[genre]
+            elif freq[genre] == max_genre:
+                max_genre_list.append(genre)
+        return ', '.join(max_genre_list)
+    return
+
+#Test wave 03 functions
+
+#Function returns list of movies that user has watched but friends haven't
+
+def get_unique_watched(user_data):
+   
+    user_titles = [movie["title"] for movie in user_data["watched"]]
+
+    friend_titles = [movie["title"] for friend in user_data["friends"] for movie in friend["watched"]]
+
+    #use set diffrence to create a list of unique titles
+    unique_titles_list = list(set(user_titles)-set(friend_titles))
+
+    #iterate through user's watched list and return a list of a movie dictionary
+    unique_movies_list = [movie for movie in user_data["watched"] for title in unique_titles_list if title == movie["title"]]
+
+    return unique_movies_list
+        
+
+
+#Function returns list of movies that friends have watched but user hasn't
+
+def get_friends_unique_watched(user_data):
+
+    unique_movies_list = []
+
+    user_titles = [movie["title"] for movie in user_data["watched"]]
+
+    friend_titles = [movie["title"] for friend in user_data["friends"] for movie in friend["watched"]]
+
+    #Use set difference to create a list of unique titles
+    unique_titles_list = list(set(friend_titles)-set(user_titles))
+
+    #iterate through user's friend's watched list and return a list of a movie dictionary
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
```

Comment:
I noticed these are the same loops as on lines 100/102. Is there a way to refactor this code to remove these duplicate loops?